packages/flutter/lib/src/widgets/focus_scope.dart

```diff
@@ -398,6 +398,7 @@ class _FocusState extends State<Focus> {
   @override
   void deactivate() {
     super.deactivate();
+    _focusAttachment?.reparent();
```

Comment:
@gspencergoog  -this makes tests pass, and the autofocus stuff will happen in `didChangeDependencies` if the node gets reactivated.

Is this expensive?  Should we be concerned about doing more work here in the case of deactivation?

I've also updated the test in framework_tests to make sure that we do in fact call `didChangeDependencies` when we deactivate an element and then reinsert it to the tree.

Reply:
This is not terribly expensive, and needs to be done, regardless.

Reply:
Can you add a comment here as to why we reparent on deactivate?

From talking to @gspencergoog: We can't unfocus or remove the node from the tree because if the widget is moved to a different part of the tree (via global key) it should retain its focus state. That's why we temporarily park it on the root focus node (via reparent) until it either gets moved to a different part of the tree (via didChangeDependencies) or until it is disposed.

Reply:
Done.